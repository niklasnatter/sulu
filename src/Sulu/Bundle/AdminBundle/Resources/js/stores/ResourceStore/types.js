// @flow
import type {IObservableValue} from 'mobx/lib/mobx';

export type ObservableOptions = {
    locale?: ?IObservableValue<string>,
};
