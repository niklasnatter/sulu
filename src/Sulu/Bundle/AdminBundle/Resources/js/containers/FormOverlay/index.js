// @flow
import FormOverlay from './FormOverlay';

export default FormOverlay;
