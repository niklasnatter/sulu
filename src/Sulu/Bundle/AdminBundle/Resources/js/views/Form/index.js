// @flow
import Form from './Form';
import toolbarActionRegistry from './registries/ToolbarActionRegistry';
import AbstractFormToolbarAction from './toolbarActions/AbstractFormToolbarAction';
import DeleteToolbarAction from './toolbarActions/DeleteToolbarAction';
import SaveWithPublishingToolbarAction from './toolbarActions/SaveWithPublishingToolbarAction';
import SaveToolbarAction from './toolbarActions/SaveToolbarAction';
import TypeToolbarAction from './toolbarActions/TypeToolbarAction';

export default Form;

export {
    toolbarActionRegistry,
    AbstractFormToolbarAction,
    DeleteToolbarAction,
    SaveWithPublishingToolbarAction,
    SaveToolbarAction,
    TypeToolbarAction,
};
