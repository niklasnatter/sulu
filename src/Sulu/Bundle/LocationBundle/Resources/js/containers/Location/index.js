// @flow
import Location from './Location';

export default Location;
