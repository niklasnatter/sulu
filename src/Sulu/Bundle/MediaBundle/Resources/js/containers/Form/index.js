// @flow
import SingleMediaUpload from './fields/SingleMediaUpload';
import MediaSelection from './fields/MediaSelection';

export {
    SingleMediaUpload,
    MediaSelection,
};
