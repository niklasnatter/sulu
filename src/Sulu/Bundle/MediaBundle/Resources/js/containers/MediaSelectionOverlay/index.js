// @flow
import MediaSelectionOverlay from './MediaSelectionOverlay';

export default MediaSelectionOverlay;
