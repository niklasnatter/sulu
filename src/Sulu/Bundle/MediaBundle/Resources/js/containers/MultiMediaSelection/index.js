// @flow
import MultiMediaSelection from './MultiMediaSelection';
import type {Value} from './types';

export default MultiMediaSelection;
export type {Value};
