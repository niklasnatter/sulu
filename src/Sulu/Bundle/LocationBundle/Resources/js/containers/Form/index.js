// @flow
import Location from './fields/Location';

export {
    Location,
};
