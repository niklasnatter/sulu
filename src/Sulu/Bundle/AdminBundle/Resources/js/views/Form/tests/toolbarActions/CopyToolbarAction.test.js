// @flow
import {mount} from 'enzyme';
import {ResourceFormStore} from '../../../../containers/Form';
import ResourceRequester from '../../../../services/ResourceRequester';
import ResourceStore from '../../../../stores/ResourceStore';
import Router from '../../../../services/Router';
import Form from '../../../../views/Form';
import CopyToolbarAction from '../../toolbarActions/CopyToolbarAction';
import conditionDataProviderRegistry from '../../../../containers/Form/registries/conditionDataProviderRegistry';

jest.mock('../../../../utils/Translator', () => ({
    translate: jest.fn((key) => key),
}));

jest.mock('../../../../stores/ResourceStore', () => jest.fn(function(resourceKey, id, observableOptions) {
    this.id = id;
    this.data = {};
    this.observableOptions = observableOptions;
    this.locale = {
        get: jest.fn(),
    };
}));

jest.mock('../../../../services/ResourceRequester', () => ({
    post: jest.fn(),
}));

jest.mock('../../../../containers/Form/stores/ResourceFormStore', () => (
    class {
        resourceStore;
        options = {};

        constructor(resourceStore) {
            this.resourceStore = resourceStore;
        }

        get id() {
            return this.resourceStore.id;
        }

        get locale() {
            return this.resourceStore.locale;
        }

        get data() {
            return this.resourceStore.data;
        }

        delete = jest.fn();
        changeMultiple = jest.fn();
    })
);

jest.mock('../../../../services/Router', () => jest.fn(function() {
    this.navigate = jest.fn();
    this.route = {
        name: 'current_route_name',
        options: {},
    };
}));

jest.mock('../../../../views/Form', () => jest.fn(function() {
    this.showSuccessSnackbar = jest.fn();
}));

function createCopyToolbarAction(options = {}) {
    const resourceStore = new ResourceStore('test');
    const formStore = new ResourceFormStore(resourceStore, 'test');
    const router = new Router({});
    const form = new Form({
        locales: [],
        resourceStore,
        route: router.route,
        router,
    });

    return new CopyToolbarAction(formStore, form, router, [], options, resourceStore);
}

test('Return item config with correct disabled, type and label', () => {
    const copyToolbarAction = createCopyToolbarAction();

    expect(copyToolbarAction.getToolbarItemConfig()).toEqual(expect.objectContaining({
        disabled: true,
        label: 'sulu_admin.create_copy',
        type: 'button',
    }));
});

test('Return  item config with enabled button if form store contains an id', () => {
    const copyToolbarAction = createCopyToolbarAction();
    copyToolbarAction.resourceFormStore.resourceStore.id = 123;

    expect(copyToolbarAction.getToolbarItemConfig()).toEqual(expect.objectContaining({
        disabled: false,
    }));
});

test('Return item config if passed visible_condition is met', () => {
    const copyToolbarAction = createCopyToolbarAction({visible_condition: '_permission.edit'});
    copyToolbarAction.resourceFormStore.resourceStore.data._permission = {edit: true};

    expect(copyToolbarAction.getToolbarItemConfig()).toEqual(expect.objectContaining({
        label: 'sulu_admin.create_copy',
    }));
});

test('Return empty item config if passed visible_condition is not met', () => {
    const copyToolbarAction = createCopyToolbarAction({visible_condition: '_permission.edit'});
    copyToolbarAction.resourceFormStore.resourceStore.data._permission = {edit: false};

    expect(copyToolbarAction.getToolbarItemConfig()).toEqual(undefined);
});

test('Include data of conditionDataProviderRegistry when evaluating passed visible_condition', () => {
    const copyToolbarAction = createCopyToolbarAction({visible_condition: '__conditionDataProviderValue'});
    expect(copyToolbarAction.getToolbarItemConfig()).toBeUndefined();

    conditionDataProviderRegistry.add(() => ({__conditionDataProviderValue: true}));
    expect(copyToolbarAction.getToolbarItemConfig()).toBeDefined();

    conditionDataProviderRegistry.clear();
    conditionDataProviderRegistry.add(() => ({__conditionDataProviderValue: false}));
    expect(copyToolbarAction.getToolbarItemConfig()).toBeUndefined();
});

test('Display confirmation dialog when button is clicked', () => {
    const copyToolbarAction = createCopyToolbarAction();
    copyToolbarAction.resourceFormStore.resourceStore.id = 3;
    // $FlowFixMe
    copyToolbarAction.resourceFormStore.resourceStore.locale.get.mockReturnValue('en');
    copyToolbarAction.resourceFormStore.options.webspace = 'sulu_io';

    const toolbarItemConfig = copyToolbarAction.getToolbarItemConfig();
    if (!toolbarItemConfig) {
        throw new Error('The toolbarItemConfig should be a value!');
    }

    const onClickCallback = toolbarItemConfig.onClick;
    if (!onClickCallback) {
        throw new Error('A onClick callback should be registered on the unpublish option');
    }

    let element = mount(copyToolbarAction.getNode());
    expect(element.instance().props).toEqual(expect.objectContaining({
        open: false,
    }));

    onClickCallback();
    element = mount(copyToolbarAction.getNode());
    expect(element.instance().props).toEqual(expect.objectContaining({
        open: true,
    }));

    expect(element.render()).toMatchSnapshot();
});

test('Close confirmation dialog when onCancel callback is fired', () => {
    const copyToolbarAction = createCopyToolbarAction();
    copyToolbarAction.resourceFormStore.resourceStore.id = 3;
    // $FlowFixMe
    copyToolbarAction.resourceFormStore.resourceStore.locale.get.mockReturnValue('en');
    copyToolbarAction.resourceFormStore.options.webspace = 'sulu_io';

    const toolbarItemConfig = copyToolbarAction.getToolbarItemConfig();
    if (!toolbarItemConfig) {
        throw new Error('The toolbarItemConfig should be a value!');
    }

    const onClickCallback = toolbarItemConfig.onClick;
    if (!onClickCallback) {
        throw new Error('A onClick callback should be registered on the unpublish option');
    }

    let element = mount(copyToolbarAction.getNode());
    expect(element.instance().props).toEqual(expect.objectContaining({
        open: false,
    }));

    onClickCallback();
    element = mount(copyToolbarAction.getNode());
    expect(element.instance().props).toEqual(expect.objectContaining({
        open: true,
    }));

    element.prop('onCancel')();
    element = mount(copyToolbarAction.getNode());
    expect(element.instance().props).toEqual(expect.objectContaining({
        open: false,
    }));
});

test('Copy resource when confirmation dialog is confirmed', () => {
    const copyPromise = Promise.resolve({id: 'copied-id', webspace: 'copied-webspace'});
    ResourceRequester.post.mockReturnValue(copyPromise);

    const copyToolbarAction = createCopyToolbarAction();
    copyToolbarAction.resourceFormStore.resourceStore.id = 3;
    // $FlowFixMe
    copyToolbarAction.resourceFormStore.resourceKey = 'pages';
    // $FlowFixMe
    copyToolbarAction.resourceFormStore.resourceStore.locale.get.mockReturnValue('en');
    copyToolbarAction.resourceFormStore.options.webspace = 'sulu_io';

    const toolbarItemConfig = copyToolbarAction.getToolbarItemConfig();
    if (!toolbarItemConfig) {
        throw new Error('The toolbarItemConfig should be a value!');
    }

    const clickHandler = toolbarItemConfig.onClick;
    if (!clickHandler) {
        throw new Error('A onClick callback should be registered on the unpublish option');
    }

    let element = mount(copyToolbarAction.getNode());
    clickHandler();

    expect(element.prop('confirmLoading')).toEqual(false);
    element.prop('onConfirm')();
    element = mount(copyToolbarAction.getNode());
    expect(element.prop('confirmLoading')).toEqual(true);
    expect(ResourceRequester.post).toBeCalledWith(
        'pages',
        undefined,
        {action: 'copy', id: 3, webspace: 'sulu_io'}
    );

    return copyPromise.then(() => {
        element = mount(copyToolbarAction.getNode());
        expect(copyToolbarAction.form.showSuccessSnackbar).toBeCalledWith();
        expect(element.prop('confirmLoading')).toEqual(false);
        expect(copyToolbarAction.router.navigate).toBeCalledWith(
            'current_route_name',
            {id: 'copied-id', webspace: 'copied-webspace'}
        );
    });
});
