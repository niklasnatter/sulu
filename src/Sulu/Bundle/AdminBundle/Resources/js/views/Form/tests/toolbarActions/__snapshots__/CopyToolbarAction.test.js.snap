// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display confirmation dialog when button is clicked 1`] = `
Array [
  <div
    class="backdrop visible fixed"
    data-testid="backdrop"
    role="button"
  />,
  <div
    class="dialogContainer open"
  >
    <div
      class="dialog"
    >
      <section
        class="content"
      >
        <div
          class="snackbar"
        >
          <div
            class="snackbar error"
            role="button"
          >
            <span
              aria-label="su-exclamation-triangle"
              class="su-exclamation-triangle icon"
            />
            <div
              class="text"
            >
              <strong>
                sulu_admin.error
              </strong>
               - 
            </div>
          </div>
        </div>
        <header
          class="header"
        >
          <span
            class="headerItem"
          >
            sulu_admin.create_copy
          </span>
        </header>
        <article
          class="article center"
        >
          sulu_admin.copy_dialog_description
        </article>
        <footer
          class="footer"
        >
          <button
            class="button primary hasText"
            type="button"
          >
            <span
              class="buttonText"
            >
              sulu_admin.ok
            </span>
          </button>
          <button
            class="button secondary hasText"
            type="button"
          >
            <span
              class="buttonText"
            >
              sulu_admin.cancel
            </span>
          </button>
        </footer>
      </section>
    </div>
  </div>,
]
`;
